Dockerfile.crystal

```diff
@@ -0,0 +1,53 @@
+# This dockerfile expects proxies to be set via --build-arg if needed
+# It also expects to be contained in the /navigation2 root folder for file copy
+# Example build command:
+# export CMAKE_BUILD_TYPE=Debug
+# export BUILD_SYSTEM_TESTS=False
+# docker build -t nav2:latest --build-arg BUILD_SYSTEM_TESTS --build-arg CMAKE_BUILD_TYPE ./
+FROM osrf/ros:crystal-desktop
```

Comment:
Any particular reason for choosing `osrf/ros:crystal-desktop`? Trying to maximize of installed dependency caching for quicker builds I presume, or is it helpful to have a desktop release install for debugging/proofing with other ros2 packages?

Could start from a leaner image base image to try and catch undeclared dependencies sooner. The CI job includes a lot of packages in the nightly fat archive, so test building only `FROM ros:crystal` if it service most of your nominal dependencies could compliment that.

Just a suggestion, I'm not sure how many of your dependencies are missing from `ros-base` meta package, but the other thing is that once you release into desktop, you'll want to make sure your testing navigation2 source build from the workspace and not the navigation2 debian that desktop may rope in. I've previously has some odd issues with sourcing colcon overlays and python ros2 package, where ros2 cli was still using the wrong python entrypoint to the installed package instead.

Reply:
> Any particular reason for choosing osrf/ros:crystal-desktop?
No. I just didn't know about the other one. Thanks for pointing it out and thanks for the other fixes. I'll update and test.